consensus/ethash/algorithm.go

```diff
@@ -174,7 +174,7 @@ func generateCache(dest []uint32, epoch uint64, seed []byte) {
 			case <-done:
 				return
 			case <-time.After(3 * time.Second):
-				logger.Info("Generating ethash verification cache", "percentage", atomic.LoadUint32(&progress)*100/uint32(rows)/4, "elapsed", common.PrettyDuration(time.Since(start)))
+				logger.Info("Generating ethash verification cache", "percentage", atomic.LoadUint32(&progress)*100/uint32(rows)/(cacheRounds+1), "elapsed", common.PrettyDuration(time.Since(start)))
```

Comment:
Hmm, shouldn't it be `cacheRounds` instead of `cacheRounds+1`?

Reply:
There is a initialization phase just before the `cacheRounds` to fill the cache with keccak512.

Reply:
Ah, indeed.